features/networking/sctp.feature

```diff
@@ -33,23 +32,257 @@ Feature: SCTP related scenarios
+      | namespace        | <%= project.name %>                                                      |
+      | oc_opts_end      |                                                                          |
+      | exec_command     | sh                                                                       |
+      | exec_command_arg | -c                                                                       |
+      | exec_command_arg | /usr/bin/nc -v <%= cb.serverpod_ip %> 30102 --sctp  <<< 'test-openshift' |
     Then the step should succeed
     And the output should contain:
       | Connected to <%= cb.serverpod_ip %>:30102 |
-      | 15 bytes sent                             |
\ No newline at end of file
+      | 15 bytes sent                             |
+    """"
+
+  # @author weliang@redhat.com
+  # @case_id OCP-28758
+  @admin
+  @destructive
+  Scenario: Expose SCTP ClusterIP Services
+    Given I store the workers in the :workers clipboard
+    And I store the number of worker nodes to the :num_workers clipboard
+    
+    Given I install machineconfigs load-sctp-module
+    And I wait up to 800 seconds for the steps to pass:
+    """
+    Given I check load-sctp-module in <%= cb.num_workers %> workers
+    """
+    
+    Given I have a project
+    Given I obtain test data file "networking/sctp/sctpserver.yaml"
+    When I run oc create as admin over "sctpserver.yaml" replacing paths:
+      | ["metadata"]["namespace"] | <%= project.name %>       |
+      | ["spec"]["nodeName"]      | <%= cb.workers[0].name %> |
+    Then the step should succeed
+    And the pod named "sctpserver" becomes ready
+
+    Given I obtain test data file "networking/sctp/sctpclient.yaml"
+    When I run oc create as admin over "sctpclient.yaml" replacing paths:
+      | ["metadata"]["namespace"] | <%= project.name %>       |
+      | ["spec"]["nodeName"]      | <%= cb.workers[1].name %> |
+    Then the step should succeed
+    And the pod named "sctpclient" becomes ready
+   
+    Given I obtain test data file "networking/sctp/sctpservice.yaml"
+    When I run oc create as admin over "sctpservice.yaml" replacing paths:
+      | ["metadata"]["namespace"] | <%= project.name %>       |
+    Given I wait for the "sctpservice" service to become ready
+    Given I use the "sctpservice" service
+    And evaluation of `service.ip(user: user)` is stored in the :service_ip clipboard
+
+    # sctpserver pod start to wait for sctp traffic
+    When I run the :exec background client command with:
+      | pod              | sctpserver                                |
+      | namespace        | <%= project.name %>                       |
+      | oc_opts_end      |                                           |
+      | exec_command     | sh                                        |
+      | exec_command_arg | -c                                        |
+      | exec_command_arg | /usr/bin/nc -l 30102 --sctp -k -m 5 -w 60 |
```

Comment:
can we use `service.port(user: user)` so we don't have to hard-code the port?

Reply:
@rbbratta  30102 is SCTP port not service port, and it is statically defined in the pod yaml file, so it's OK by using the hard-code sctp port number. Thanks!